elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/ElasticsearchJavaRestClient.scala

```diff
@@ -14,20 +17,51 @@ class ElasticsearchJavaRestClient(client: RestClient) extends HttpRequestClient
 
   import scala.collection.JavaConverters._
 
+  private def fromResponse(r: org.elasticsearch.client.Response): HttpResponse = {
+    val entity = Option(r.getEntity).map { entity =>
+      val contentEncoding =
+        Option(entity.getContentEncoding).map(_.getValue).getOrElse("UTF-8")
+      implicit val codec = Codec(Charset.forName(contentEncoding))
+      val body           = Source.fromInputStream(entity.getContent).mkString
+      HttpEntity.StringEntity(body, Some(contentEncoding))
+    }
+    val headers = r.getHeaders.map { header =>
+      header.getName -> header.getValue
+    }.toMap
+    HttpResponse(r.getStatusLine.getStatusCode, entity, headers)
+  }
+
+  private def doAsync[F[_]: AsyncExecutor](f: ResponseListener => Unit): F[HttpResponse] = {
+    AsyncExecutor[F].async { cb =>
```

Comment:
Is this just a pattern?

Could you do,

```scala
private def doAsync[F[_]](f: ResponseListener => Unit)(implicit executor: AsyncExecutor[F[_]]): F[HttpResponse] = {
  executor.async { cb =>
````

Reply:
Yep that would work fine -- I have no preference. Would you like me to do that uniformly throughout the PR?